code/_helpers/type2type.dm

```diff
@@ -96,6 +97,19 @@
 		if ("SOUTHEAST") return 6
 		if ("SOUTHWEST") return 10
 
+// Turns a direction into text showing all bits set
+/proc/dirs2text(direction)
+	if(!direction)
+		return ""
+	var/list/dirs = list()
+	if(direction & NORTH) dirs += "NORTH"
+	if(direction & SOUTH) dirs += "SOUTH"
+	if(direction & EAST) dirs += "EAST"
+	if(direction & WEST) dirs += "WEST"
+	if(direction & UP) dirs += "UP"
+	if(direction & DOWN) dirs += "DOWN"
+	return dirs.Join(" ")

```

Comment:
```
// Turns a direction into text showing all bits set
/proc/dirs2text(direction)
    . = ""
    if(direction & [DIR]) . += "DIR "
    [...]
    . = copytext(., 1, (.).len - 1)
```

Reply:
This code makes no sense. DIR is an integer and copytext and repeated string operations aren't an improvement on Join().